analyses/clone-phylogeny-analysis/06A-create-numbat-plots.Rmd

```diff
@@ -0,0 +1,479 @@
+
+p_cnv <- ggplot(plot_data_p_cnv, aes(x = UMAP_1, y = UMAP_2, color = Feature)) +
+  geom_point(alpha = 0.8) +  # Adjust transparency if needed
+  scale_color_gradient2(low = 'royalblue', mid = 'white', high = 'red3', midpoint = 0.5, name = 'Posterior') +
+  ggtitle("Allele") +
+  theme_Publication () +
+  theme(legend.position = "right")
+
+########################################################################
+## Expression-based posterior (p_cnv_x)
+# Extract feature data from metadata
+feature_data <- metadata_table$p_cnv_x
+
+# Combine UMAP coordinates and feature data into a data frame
+plot_data <- data.frame(UMAP_1 = umap_data[, 1], UMAP_2 = umap_data[, 2], Feature = feature_data)
+
+p_cnv_x <- ggplot(plot_data, aes(x = UMAP_1, y = UMAP_2, color = Feature)) +
+  geom_point(alpha = 0.8) +  # Adjust transparency if needed
+  scale_color_gradient2(low = 'royalblue', mid = 'white', high = 'red3', midpoint = 0.5, name = 'Posterior') +
+  ggtitle("Expression") +
+  theme_Publication () +
+  theme(legend.position = "right")
+########################################################################
+## Allele-based posterior (p_cnv_y)
+# Extract feature data from metadata
+feature_data <- metadata_table$p_cnv_y
+
+# Combine UMAP coordinates and feature data into a data frame
+plot_data <- data.frame(UMAP_1 = umap_data[, 1], UMAP_2 = umap_data[, 2], Feature = feature_data)
+
+p_cnv_y <- ggplot(plot_data, aes(x = UMAP_1, y = UMAP_2, color = Feature)) +
+  geom_point(alpha = 0.8) +  # Adjust transparency if needed
+  scale_color_gradient2(low = 'royalblue', mid = 'white', high = 'red3', midpoint = 0.5, name = 'Posterior') +
+  ggtitle("Joint") +
+  theme_Publication () +
+  theme(legend.position = "right")
+```  
+
+```{r plot-4, fig.width = 20, fig.height = 6, fig.fullwidth = TRUE, echo=TRUE}
+fname <- paste0(numbat_plots_dir, "/", sample_name[i], "/", "plot4.png")
+p <- print(p_cnv + p_cnv_x + p_cnv_y + plot_layout(ncol = 3))
+ggsave(file = fname, p, width = 20, height = 6, device = "png")
+```
+
+# CNVs and seg
+
+```{r plot-5, fig.width = 20, fig.height = 8, fig.fullwidth = TRUE, echo=TRUE}
+split_data <- metadata_table$seg  # Replace 'seg' with your splitting variable
+
+# Combine UMAP coordinates, feature data, and the split variable into a data frame
+plot_data <- data.frame(UMAP_1 = umap_data[, 1], UMAP_2 = umap_data[, 2], Feature = feature_data_p_cnv, Seg = split_data)
+
+# Create the plot using ggplot2
+p1 <- ggplot(plot_data, aes(x = UMAP_1, y = UMAP_2, color = Feature)) +
+  geom_point(alpha = 0.8) +  # Adjust transparency if needed
+  scale_color_gradient2(low = 'royalblue', mid = 'white', high = 'red3', midpoint = 0.5, name = 'Posterior') +
+  facet_wrap(~Seg) +  # Split by 'seg' (or your splitting variable)
+  theme_Publication () +
+  theme(legend.position = "right")
+
+p2 <- ggplot(plot_data_p_cnv, aes(x = UMAP_1, y = UMAP_2, color = Feature)) +
```

Comment:
This plot seems to be a duplicate of the current first plot in the `Joint posterior probability of the CNV` plots. And again, I think it is displaying too much overlapping data to be informative.